lib/error_tracker.ex

```diff
@@ -118,7 +118,7 @@ defmodule ErrorTracker do
     {kind, reason} = normalize_exception(exception, stacktrace)
     {:ok, stacktrace} = ErrorTracker.Stacktrace.new(stacktrace)
     {:ok, error} = Error.new(kind, reason, stacktrace)
-    context = Map.merge(get_context(), given_context)
+    context = get_context() |> Map.merge(given_context) |> filter_context_data()
 
     if enabled?() && !ignored?(error, context) do
       {_error, occurrence} = upsert_error!(error, stacktrace, context, reason)
```

Comment:
I think that it may make sense to filter the context data only after we are sure that we want to track the error.

This would reduce the number of operations that the ErrorTracker does for ignored errors, which I think is important given that this calls user-defined code that we don't have any control over.

```suggestion
    context = Map.merge(get_context(), given_context)

    if enabled?() && !ignored?(error, context) do
      filtered_context = filter_context_data(context)
      {_error, occurrence} = upsert_error!(error, stacktrace, filtered_context, reason)
```

This would mean that the ignorer, if defined, has access to the full unfiltered context but I think that would be OK as long as we document it.

What do you think @odarriba?

Reply:
I agree, as that function may do heavy calculations (or not, who knows)